### ISSUE TITLE
Running Batch Jobs

### ISSUE THREAD

**Issue by kaufmann42** [pygridtools/drmaa-python#38]
Hi I'm trying to run a batch job utilizing the --array slurm option. Wondering if this is possible using drmaa-python. I know there is a runBulkJobs(...), however it seems that this doesn't run an array of jobs. There doesn't seem to be any $SLURM_ARRAY_TASK_ID (or the likes) associated with that run environment.

```
with drmaa.Session() as s:
                try:
                    # create job template
                    jt = s.createJobTemplate()
                    jt.nativeSpecification='--mem-per-cpu='+self.memory + ' --array=1-3' + ' --time=' + self.time
                    jt.remoteCommand = command
                    print(jt.nativeSpecification)

                    # run job
                    joblist = s.runBulkJobs(jt, 1, 3, 1)

                    # wait for the return value
                    s.synchronize(joblist, self.convertToSeconds(), False)

                    for curjob in joblist:
                        print('Collecting job ' + curjob)
                        retval = s.wait(curjob, drmaa.Session.TIMEOUT_WAIT_FOREVER)
                        print('Job: {0} finished with status {1} and was aborted {2}'.format(retval.jobId, retval.exitStatus, retval.wasAborted))
                        if retval.wasAborted == True:
                            print("Ran out of memeory using: " + self.memory)
                            self.increaseMemory(6000)
                        # if zero exit code then break and job is over
                        elif retval.exitStatus == 0:
                            break
                except drmaa.ExitTimeoutException:
                    print("Ran out of time using: " + self.time)
                    self.increaseTime(6)
                except drmaa.OutOfMemoryException:
                    print("Ran out of memeory using: " + self.memory)
                    self.increaseMemory(6000)
```

aside: I'm also having trouble with it throwing a OutOfMemoryException.. Therefore am forced to assume it was aborted due to memory (not preferable) so advice on what's happening there would be great.

Thanks!

**Comment by Unode** [comment]
I also see the segfault on SLURM. Possibly related to [this](https://groups.google.com/forum/#!topic/slurm-devel/sGS_H4Bx4BQ).

**Comment by unode** [comment]
TL;DR: In my case this is no longer a problem with `drmaa-python`.

Tried to debug the code to figure out which call triggered the segfault.
At the moment I'm not sure that what I'm seeing is the same error reported above.

The problem seems to be in the [call to drmaa_get_next_job_id](https://github.com/pygridtools/drmaa-python/blob/master/drmaa/helpers.py#L288).
The job is submitted and queued, it reaches the `while` loops, loops once and a `jid` is collected. Printing `jid` returns the same id as the submitted job. Yet as soon as the while condition is evaluated a second time it segfaults.

Afterwards I tried compiling the latest [libdrmaa](http://apps.man.poznan.pl/trac/slurm-drmaa/downloads) for SLURM and with it I could no longer reproduce the segfault.
This is also true even when compiling manually the same libdrmaa.so version provided system-wide.

It seems like this is an upstream problem and only seems to affect some releases when a certain combination of conditions is met.

**Comment by unode** [comment]
For anyone that lands on this post after dealing with segmentation fault errors on SLURM, you might want to ask your cluster administrator to use `libdrmaa.so` from https://github.com/FredHutch/slurm-drmaa.

It's far from perfect and will still segfault if there are options in the `nativeSpecification` string that it cannot parse. Regardless I've managed to workaround most of the issues mentioned above with this version.

**Comment by jakirkham** [comment]
Perhaps worth adding a note in the docs and closing as this is a DRMAA implementation issue.

**Comment by unode** [comment]
Just adding that https://github.com/natefoo/slurm-drmaa/commit/7b5991efc03ab14fdc9e7af67dc91f6085e4d648 solves this issue.
I agree with @jakirkham.
